packages/vite/src/node/optimizer/scan.ts

```diff
@@ -85,33 +84,23 @@ export async function scanImports(
     debug(`Crawling dependencies using entries:\n  ${entries.join('\n  ')}`)
   }
 
-  const tempDir = path.join(config.cacheDir!, 'temp')
   const deps: Record<string, string> = {}
   const missing: Record<string, string> = {}
   const container = await createPluginContainer(config)
   const plugin = esbuildScanPlugin(config, container, deps, missing, entries)
 
-  await Promise.all(
-    entries.map((entry) =>
-      build({
-        entryPoints: [entry],
-        bundle: true,
-        format: 'esm',
-        logLevel: 'error',
-        outdir: tempDir,
-        plugins: [plugin]
-      })
-    )
-  )
-
-  try {
-    emptyDir(tempDir)
-    fs.rmdirSync(tempDir)
-  } catch (err) {
-    if (err.code !== 'ENOENT') {
-      throw err
-    }
-  }
+  const { entryPoints = [], plugins = [], ...esbuildOptions } =
+    config.optimizeDeps?.esbuildOptions ?? {}
+
+  await build({
+    write: false,
+    entryPoints: [...entryPoints, ...entries],
```

Comment:
Same question here about merging `entryPoints`.

Additionally, I moved it from multiple builds to a single one – should be marginally faster. Have I not considered a reason to do it separately?